File: Searching Algorithms/Binary_Search.c
/*
Implementation of binary search algorithm
*/
#include<stdio.h>
#include<stdbool.h>
int main(){
	int arr[] = {1,2,3,4,5,6,7,8,9,10};
	int high=9,low=0,mid,num;
	bool flag = false;
	printf("Please enter the number you want to search for: ");
	scanf("%d",&num);
	while(high>low){
		mid = (high+low)/2;
		if(arr[mid]==num) {flag=true;break;}
		if(arr[mid]>num) high=mid-1;
		if(arr[mid]<num) low=mid+1;
	}
	flag?printf("\nElement is present at index position %d",mid+1):printf("\nElement is not present!");
	return 0;
}

File: 02 Searching Algorithms/Binary_Search.py
#Program to implement binary search
import sys
import math

size = int(input("Please enter the size of the list: "))
myList = list()
for i in range(0,size):
    myList.append(int(input("Please enter the element at pos {}: ".format(i+1))))

num = int(input("Please enter the number you want to search for: "))
low, high = 0, size-1

while(low<=high):
    mid = math.floor(low + (high-low)/2)
    #print("Current midIdx={} and midElement={}".format(mid,myList[mid]))
    if myList[mid]>num:
        high=mid-1
    elif myList[mid]<num:
        low=mid+1
    else:
        print("Element found at index {}".format(mid))
        sys.exit()
print("Given element does not exist in the list")
        
